File: MundoPc/src/ar/com/system2023/mundopc/Computadora.java
package ar.com.system2023.mundopc;

public class Computadora {
    private final int idComputadora; // Al ser una variable constante de tipo final, solo utilizamos el get, el set NO.
    private String nombre;
    private Monitor monitor;
    private Teclado teclado;
    private Raton raton;
    private static int contadorComputadoras;
    
    // Relación de agregación: Los objetos de tipo monitor pueden existir sin necesidad de objetos de tipo Computadora. 
    
    // Constructor vacío.
    private Computadora(){
        this.idComputadora = ++Computadora.contadorComputadoras; // Incremento de computadoras.
    }
    
    // Constructor 2.
    public Computadora(String nombre, Monitor monitor, Teclado teclado, Raton raton){
        this();
        this.nombre = nombre;
        this.monitor = monitor;
        this.teclado = teclado;
        this.raton = raton;
    }

    public int getIdComputadora(){
        return this.idComputadora;
    }
    
    public String getNombre() {
        return this.nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public Monitor getMonitor() {
        return this.monitor;
    }

    public void setMonitor(Monitor monitor) {
        this.monitor = monitor;
    }

    public Teclado getTeclado() {
        return this.teclado;
    }

    public void setTeclado(Teclado teclado) {
        this.teclado = teclado;
    }

    public Raton getRaton() {
        return this.raton;
    }

    public void setRaton(Raton raton) {
        this.raton = raton;
    }

    @Override
    public String toString() { // Nos muestra la relación entre todos los métodos.
        return "Computadora{" + "idComputadora=" + idComputadora + ", nombre=" + nombre + ", monitor=" + monitor + ", teclado=" + teclado + ", raton=" + raton + '}';
    }    
}

File: python/semana4/mundo_pc/computadora.py
from mundo_pc.monitor import Monitor
from mundo_pc.raton import Raton
from mundo_pc.teclado import Teclado


class Computadora:

    contador_computadoras = 0

    def __int__(self, nombre, monitor, teclado, raton):
        Computadora.contador_computadoras += 1
        self._id_computadora = Computadora.contador_computadoras
        self._nombre = nombre
        self._monitor = monitor
        self._teclado = teclado
        self._raton =raton

    def __str__(self):
        return f'''
        {self._nombre}: {self._id_computadora}
        Monitor: {self._monitor}
        Teclado: {self._teclado}
        Raton: {self._raton}
        '''


if __name__ == '__main__':
    teclado1 = Teclado("HP", "USB")
    monitor1 = Monitor("HP", "15 Pulgadas")
    raton1 = Raton ("HP","USB")
    computadora1 = Computadora( 'HP' , monitor1 , teclado1, raton1)
    print(computadora1)

    teclado2 = Teclado("Acer", "USB")
    monitor2 = Monitor("Acer", "15 Pulgadas")
    raton2 = Raton("Acer", "USB")
    computadora2 = Computadora('Acer', monitor1, teclado1, raton1)
    print(computadora2)